tests/kafkatest/tests/tools/log_compaction_test.py

```diff
@@ -0,0 +1,69 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from ducktape.utils.util import wait_until
+from ducktape.tests.test import Test
+from ducktape.mark.resource import cluster
+
+from kafkatest.services.kafka import config_property
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.log_compaction_tool import LogCompactionTool
+
+class LogCompactionTest(Test):
+    """
+    Tests Log Compaction
+    :return: None
+    """
+    def __init__(self, test_context):
+        super(LogCompactionTest, self).__init__(test_context)
+        self.num_zk = 1
+        self.num_brokers = 1
+
+        self.zk = ZookeeperService(test_context, self.num_zk)
+        self.kafka = None
+        self.compaction_verifier = None
+
+    def setUp(self):
+        self.zk.start()
+
+    def start_kafka(self, security_protocol, interbroker_security_protocol):
+        self.kafka = KafkaService(
+            self.test_context,
+            num_nodes = self.num_brokers,
+            zk = self.zk,
+            security_protocol=security_protocol,
+            interbroker_security_protocol=interbroker_security_protocol,
+            server_prop_overides=[
+                [config_property.LOG_SEGMENT_BYTES, "1073000"],
```

Comment:
How was this number chosen?  Should there be a constant?

Reply:
Configured smaller segment size  (1MB, default is 1GB) to create more segments  quickly for compaction. moved to constant.